### ISSUE TITLE
EmguCV Unity Lag

### ISSUE THREAD

**Issue by qwenzo** [emgucv/emgucv#258]
Hello,

i was trying to use EmguCV in Unity. in the Update function, I just grab a (1920x1080) frame from VideoCapture and apply it to a sprite. currently, this method costs lots of memory for some reason and the scene is extremely laggy. 

```
void Update()
    {
        _frame = cap.QueryFrame();
        Image<Bgr, byte> currImage = _frame.ToImage<Bgr, byte>();
        //CvInvoke.Imshow("test", _frame);
        Destroy(cameraTex);
        cameraTex = TextureConvert.ImageToTexture2D<Bgr, byte>(currImage, FlipType.Horizontal);
        RenderTexture(cameraTex);
        ResizeTexture(cameraTex);

    }
 private void RenderTexture(Texture2D texture)
    {
            Image image = this.GetComponent<Image>();
            Sprite.DestroyImmediate(image.sprite);
            image.sprite = Sprite.Create(texture, new Rect(0, 0, texture.width, texture.height), new Vector2(0.5f, 0.5f));
    }

    private void ResizeTexture(Texture2D texture)
    {
        Image image = this.GetComponent<Image>();
        var transform = image.rectTransform;
        transform.sizeDelta = new Vector2(texture.width, texture.height);
        //transform.position = new Vector3(-texture.width / 2, -texture.height / 2);
        transform.anchoredPosition = new Vector2(0, 0);
    }
```

**Comment by johnjake** [comment]
private void InitializedCamera()
          {
               _capture = new VideoCapture(UtilConstant.haarRemoteCamera);
               _capture.SetCaptureProperty(CapProp.Fps, 55);
               _capture.ImageGrabbed += ActivateCamera;
                _capture.Start();
            }


         private void ActivateCamera(object sender, EventArgs e)
          { 
            _capture.Retrieve(_frameMat);
                var currentFrame = _frameMat.ToImage<Bgr, byte>().Resize(1080, 720, Emgu.CV.CvEnum.Inter.Cubic);
                var faceDetect = new CascadeClassifier(UtilConstant.haarFileFace);
                var facesDetected = faceDetect.DetectMultiScale(currentFrame, 1.2,
                10,
                   new Size(100, 100),
                Size.Empty);
            foreach (var faceRectangle in facesDetected)
            {
                currentFrame.Draw(faceRectangle, new Bgr(Color.Red), 2);
            }


                imgBox.Image = currentFrame.ToBitmap();
           }

**Comment by qwenzo** [comment]
@johnjake

i get this error for some reason " error CS0012: The type 'ExceptionHandler' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.ServiceModel, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
"

**Comment by johnjake** [comment]
@qwenzo what version or release are you using? I am using the latest emgucv from nuget.

**Comment by emgucv** [comment]
@qwenzo is using Unity. 

I would recommend checking out the CameraScene included in the official Emgu CV v4.1 for Unity commercial release. For true cross platform application, you should use Unity's WebCamTexture to capture frames instead of using the VideoCapture class, which only works for Desktop and server, but not for mobile devices. FYI, Open CV do not implement VideoCapture for iOS or Android.

If you really need to use VideoCapture class as in the first post, changing:
```
_frame = cap.QueryFrame();
Image<Bgr, byte> currImage = _frame.ToImage<Bgr, byte>();
//CvInvoke.Imshow("test", _frame);
Destroy(cameraTex);
cameraTex = TextureConvert.ImageToTexture2D<Bgr, byte>(currImage, FlipType.Horizontal);
```
to 
```
if (_frame == null)
  _frame = new Mat();
cap.Grab()
cap.Retrieve(_frame);
using(image<Bgr, byte> currImage = _frame.ToImage<Bgr, byte>())
{
  //CvInvoke.Imshow("test", _frame); 
  Destroy(cameraTex);
  cameraTex = TextureConvert.ImageToTexture2D<Bgr, byte>(currImage, FlipType.Horizontal);
}
```
Should help keep the memory usage down. It will re-use Mat memory instead of creating new one for every frame, and release Image<,> objects as soon as it is no longer used, rather than relying on Garbage collector to release the image memory.

**Comment by emgucv** [comment]
Closing ticket now.
